src/SegmentedArray.chpl

```diff
@@ -134,7 +134,14 @@ module SegmentedArray {
         end = offsets.a[idx+1] - 1;
       }
       // Take the slice of the bytearray and "cast" it to a chpl string
-      var s = interpretAsString(values.a[start..end]);
+//      var s = interpretAsString(values.a[start..end]);
+      var tmp=values.a[start..end];
```

Comment:
I'm not deeply familiar with this code, so take the following performance note with a grain of salt:  Assuming that `start..end` typically only involve 1 or 2 locales, I suspect it would be better to write this either as:

```chapel
var tmp=values.a[{start..end}];
```

which will have the effect of making `tmp`'s a local array with the domain `{start..end}` rather than a distributed array aligned with elements `start..end` from `values.a`.  The key rule here is that arrays sliced with a range use the original domain intersected with the range, preserving the original domain's distribution (in this case, a block-distributed array I believe).  Whereas an array sliced with a domain will result in an array defined by that slicing domain.  The `interpretAsString()` routine seems to have done this localization as well, in a different way.

Another way to achieve this would be to give `tmp` an explicit local array type, like:

```chapel
var tmp: [start..end] uint(8) = values.a[start..end];
```

I haven't checked this rewrite for correctness at all.

Reply:
Thanks for the comments and suggestions! I added the class SegSArray class based on the the SegString class. Most of the codes are the same.  I'd like the developer of SegString join to discuss the problem.

Reply:
Isn't this code change contained within the `SegString` class itself, though?  Maybe you could say more about what motivated the change?

Reply:
I did not change the codes in SegString. Let me check more about it.

Reply:
> Thanks for the comments and suggestions! I added the class SegSArray class based on the the SegString class. Most of the codes are the same. I'd like the developer of SegString join to discuss the problem.

@reuster986 is the developer of chapel `SegString` class and the python `Strings` class, I also tagged him as a reviewer.